rmf_rtls/rmf_rtls/rmf_rtls_server.py

```diff
@@ -0,0 +1,269 @@
+
+# ORM interface
+import rmf_rtls.database as db
+import rmf_rtls.utils as utils
+
+from pydantic import BaseModel, Field, parse_obj_as
+
+app = FastAPI()
+
+tag_state_schema = schemas.rtls_tag_state()
+RtlsTagStateModel = rtls_tag_state.RtlsTagState
+Transformation2DModel = transformation_2D.Transformation2D
+
+
+###############################################################################
+class RtlsTagStatesResponse(BaseModel):
+    success: bool
+    response: str = Field(
+        ..., description='response msg'
+    )
+    data: Optional[List[RtlsTagStateModel]] = Field(
+        None, description='rtls tag states data if avail'
+    )
+
+
+class Transfromation2DResponse(BaseModel):
+    success: bool
+    response: str = Field(
+        ..., description='response msg'
+    )
+    data: Optional[Transformation2DModel] = Field(
+        None, description='transformation of 2 maps if avail'
+    )
+
+
+###############################################################################
+
+@app.post('/open-rmf/rtls/tag_state')
+async def update_tag_state(dest: RtlsTagStateModel):
+    """
+    Set Tag States
+    """
+    tag_state_orm = db.RtlsTagState.from_pydantic(dest)
+    return await tag_state_orm.save()
+
+
+@app.get('/open-rmf/rtls/tag_state/',
+         response_model=RtlsTagStatesResponse)
+async def get_tag_state(
+    tag_id: Optional[str] = None,
+    asset_type: Optional[str] = None,
+    asset_subtype: Optional[str] = None,
+):
+    """
+    Get matching tag states
+    """
+    # remove all input args Keys that contain None value
+    filter_params = {k: v for k, v in locals().items() if v is not None}
+    # raw way to convert pk to commmon name: 'id'
+    if "tag_id" in filter_params:
+        filter_params['id'] = filter_params.pop("tag_id")
+    print("filter parasms ::", filter_params)
+
+    tag_states = await db.TtmRtlsTagState.filter(**filter_params)
+    return RtlsTagStatesResponse(
+        success=True,
+        response="Successfully query matching tag states",
+        data=[db.RtlsTagState.from_ttm(t) for t in tag_states]
+    )
+
+
+@app.delete("/open-rmf/rtls/tag_state/")
+async def delete_tag_state(tag_id: str):
+    """
+    Delete a map transformation
+    """
+    tf_2d = await db.TtmRtlsTagState.get_or_none(id=tag_id)
+    if tf_2d is None:
+        raise HTTPException(status_code=404, detail="Item not found")
+    await tf_2d.delete()
+    return RtlsTagStatesResponse(
+        success=True,
+        response=f"Successfully tag_id {tag_id}")
+
+###############################################################################
+
+
+@app.post('/open-rmf/rtls/map_transformation/')
+async def update_map_transformation(dest: Transformation2DModel):
+    """
+    Set map transformation
+    """
+    # TODO: a check to ensure each: target_map is unique
+    tf_2d_orm = db.Transformation2D.from_pydantic(dest)
+    return await tf_2d_orm.save()
+
+
+@app.get('/open-rmf/rtls/map_transformation/',
```

Comment:
is this supposed to get transformation of only 1 map, or do we need to handle multiple maps?

Reply:
Now it is able to create a "TF tree", in accordance with the provided transformations. 
E.g. if user provides transformation of A->B, B->C, B->D. 
```
E.g.  A->B->C
         |->D
```
you will be able to query, A->D transformation using BFS